From 6af16ef651107e73fa09be7b3c9d2cf93fa15259 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Mon, 21 Mar 2016 13:32:34 +0100
Subject: [PATCH] cpufreq_hotplug: Revert Amlogic patch to fix plugging last
 CPU core

With Amlogic patch the last core would very rarely go online. Revert
this patch to bump performance in high load cases.
---
 drivers/cpufreq/cpufreq_hotplug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/cpufreq_hotplug.c b/drivers/cpufreq/cpufreq_hotplug.c
index 244aa07..36332fc 100755
--- a/drivers/cpufreq/cpufreq_hotplug.c
+++ b/drivers/cpufreq/cpufreq_hotplug.c
@@ -608,7 +608,7 @@ static int __ref cpu_hotplug_thread(void *data)
 		if(*hotplug_flag == CPU_HOTPLUG_PLUG){
 			*hotplug_flag = CPU_HOTPLUG_NONE;
 			j = 0;
-			for(i = 0; (i < max_cpu_num) && (num_online_cpus() < max_cpu_num); i++){
+			for(i = 0; i < max_cpu_num; i++){
 				if(cpu_online(i))
 					continue;
 				j++;
-- 
1.8.3.1

